game/script-topics.rpy

```diff
@@ -1607,6 +1607,226 @@ label talk_sustainable_fashion:
 
     return
 
+# Natsuki gets a nickname from the player, assuming they aren't blocked from doing so
+init 5 python:
+    registerTopic(
+        Topic(
+            persistent._topic_database,
+            label="talk_give_nickname",
+            unlocked=True,
+            prompt="Can I give you a nickname?",
+            conditional="persistent.jn_player_can_nickname_natsuki",
+            category=["Natsuki"],
+            player_says=True,
+            affinity_range=(jn_affinity.ENAMORED, jn_affinity.LOVE),
+            location="classroom"
+        ),
+        topic_group=TOPIC_TYPE_NORMAL
+    )
+
+label talk_give_nickname:
+    # Natsuki hasn't been nicknamed before
+    if persistent.jn_player_natsuki_current_nickname is None:
+        n "Eh?{w=0.2} You want to give me a nickname?"
+        n "Why?{w=0.2} Natsuki not good enough for you?{w=0.2} Is that it?"
+        n "Huh?{w=0.2} Come on, [player]!{w=0.2} Spit it out!"
+        n "..."
+        n "Relax,{w=0.1} [player]!{w=0.2} Jeez!{w=0.2} I'm just kidding!"
+        n "Ehehe."
+        n "Well...{w=0.3} I don't see why not!" 
+
+    # Another nickname is being assigned
+    else:
+    
+        # Account for strikes
+        if persistent.jn_player_nicknames_bad_given_total == 0:
+            n "Oh?{w=0.2} You wanna give me another nickname?"
+            n "Sure,{w=0.1} why not!"
+
+        elif persistent.jn_player_nicknames_bad_given_total == 1:
+            n "You want to give me a new nickname?"
+            n "Alright,{w=0.1} [player]."
+
+        elif persistent.jn_player_nicknames_bad_given_total == 2:
+            n "Another nickname,{w=0.1} [player]?{w=0.2} Fine."
+            n "Just...{w=0.3} think a little about what you choose,{w=0.1} 'kay?"
+
+        elif persistent.jn_player_nicknames_bad_given_total == 3:
+            n "Alright,{w=0.1} [player]."
+            n "Just remember.{w=0.3} You've had your final warning about this."
+            n "Don't let me down again."
+
+    # Validate the nickname, respond appropriately
+    $ nickname = renpy.input(prompt="What did you have in mind,{w=0.2} [player]?", allow="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-' ", length=30).strip()
```

Comment:
Can we make a const for the `allow` values? I figure they might see use elsewhere so it'd be cleaner to just refer to the constant